src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @fileoverview This rule checkes header strict-transport-security is sent for resources served over HTTPS, and max-age has proper  value.
+ */
+
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+import { isDataURI, isHTTPS, normalizeString } from '../../utils/misc'; // eslint-disable-line no-unused-vars
+import { IAsyncHTMLElement, IResponse, IFetchEnd, IRule, IRuleBuilder } from '../../types'; // eslint-disable-line no-unused-vars
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The minimum period (in seconds) allowed for `max-age`. */
+        let minimum: number;
+
+        const loadRuleConfigs = () => {
+            minimum = (context.ruleOptions && context.ruleOptions.minimum) || 10886400;
+        };
+
+        const maxAgeLessThan = (headerValue: string, limit: number) => {
+            const maxAge = headerValue.match(/max-age=([^;]+);|max-age=([^;]+)$/).pop();
```

Comment:
> One question though - the doc says Directive names are case-insensitive. So that means MAX-AGE=... is not allowed and should fail the test, right? 

No, it should be allowed. As you said they are case-**insensitive**, so it doesn't matter if it's `max-age`, `MAX-AGE`, or `mAx-AgE`, they are all valid.

Also, the above is the reason why I add [these types of tests](https://github.com/sonarwhal/sonar/blob/7cd550c353141a25f9f8530fabd78b3cecf9f97f/tests/lib/rules/content-type/tests.ts#L266-L273)), and you should too.

> Also, It mentioned `The order of appearance of directives is not significant`, so if we split, it's possible that max-age is not the first element. Am I understanding it correctly?

Yes.

---

Other things, the rule should check for other things, such as:

> [`The REQUIRED "max-age" directive`...](https://tools.ietf.org/html/rfc6797#section-6.1.1)

Check that the required directives are provided.

> [`All directives MUST appear only once in an STS header field. Directives are either optional or required, as stipulated in their definitions.`](https://tools.ietf.org/html/rfc6797#section-6.1)

Check that directives only appear once.

Question: What do browser do if you have multiple `max-age` directives? Do they follow the spec?